### DISCUSSION TITLE
taro 多页面，或者能否给一个完整的demo？，相关资料比较少

### DISCUSSION THREAD

**Discussion by zouchao1** [NervJS/taro#15168] [Q & A]
### 这个特性解决了什么问题？
**根据文档**
```
 router: {
      mode: 'multi', 
    }, 
``` 
   **开启多页面应用，想要以每个page文件夹下的template index.html作为模版，并且每个页面他自己的子路由如何单独创建**
```   
webpackChain(chain, webpack) {
      chain.plugins.delete('html');

      chain.merge({
        plugin: {
          install: {
            plugin: require('html-webpack-plugin'),
            args: [
              {
                template: path.resolve(__dirname, '..', 'src/pages/home/index.html'),
                filename: 'pages/home/home.html',
              },
              {
                template: path.resolve(__dirname, '..', 'src/pages/about/index.html'),
                filename: 'pages/about/about.html',
              },
            ],
          },
        },
      });
    },
  ```
   **发现只会打包后dist文件下的pages生成一个home文件夹里面有一个home.html**
      

### 这个 API 长什么样？
提供一个配置参数开启多页应用后有自己的模版和子路由等

**Comment by zouchao1** [comment]
<img width="200" alt="截屏2024-01-24 21 19 10" src="https://github.com/NervJS/taro/assets/35158133/ada0e3f6-70b9-41c4-beef-464413030402">

**Comment by ZakaryCode** [comment]
多页路由本身只有一个配置就可以开启，并不需要单独示例项目

**Comment by ZakaryCode** [comment]
> 开启多页面应用，想要以每个page文件夹下的template index.html作为模版，并且每个页面他自己的子路由如何单独创建

以 `pages/home/index` 为例，可以参考以下配置：

```js
chain.plugin('pages/index/index').tap(args => {
  args[0].template = path.resolve(path.dirname(args[0].template), 'pages/index/index.html')
  return args
})
```

**Reply by zouchao1** [reply]
<img width="780" alt="截屏2024-01-25 11 29 53" src="https://github.com/NervJS/taro/assets/35158133/158eb542-8af1-4cfb-bed8-b16d418982f2">
解决

**Comment by yangliguo7** [comment]
@zouchao1 能看下你的例子吗？

@ZakaryCode 我想在taro运行的时候，在本地额外的运行一个单独的html文件，类似多入口。这种怎么操作

**Reply by zouchao1** [reply]
上面的图片就是每个页面有自己的html呀，然后配置route开启属性就可以了
